app/frontend/src/components/EditLocationMap.tsx

```diff
@@ -0,0 +1,132 @@
+import { Box, BoxProps, makeStyles } from "@material-ui/core";
+import { LatLng } from "leaflet";
+import React, { useMemo, useRef, useState } from "react";
+import { Circle, Marker, useMapEvents } from "react-leaflet";
+import { userLocationMaxRadius, userLocationMinRadius } from "../constants";
+import { User } from "../pb/api_pb";
+import Map from "./Map";
+
+const useStyles = makeStyles({
+  root: {},
+});
+
+export interface UserLocation {
+  lat: number;
+  lng: number;
+  //meters
+  radius: number;
+}
+
+export interface EditLocationMapProps extends BoxProps {
+  user: User.AsObject;
+  setLocation: (location: UserLocation) => void;
+}
+
+export default function EditLocationMap({
+  user,
+  setLocation,
+  ...otherProps
+}: EditLocationMapProps) {
+  const classes = useStyles();
+
+  const center = useMemo(() => new LatLng(user.lat, user.lng), [user]);
+  //What type is this useRef supposed to be!? Marker doesn't want to work
+  const markerRef = useRef<any>(null);
```

Comment:
Couldn't get typing to work here

Reply:
Me neither. I think the correct typing is `typeof Marker` but then in next lines `getLatLng` is not accepted by Typescript.